ipfx/feature_record.py

```diff
@@ -0,0 +1,136 @@
+import pandas as pd
+import numpy as np
+
+
+def build_cell_feature_record(cell_features, sweep_features):
+    ephys_features = {}
+
+    # find hero and rheo sweeps in sweep table
+    rheo_sweep_num = cell_features["long_squares"]["rheobase_sweep"]["id"]
+    rheo_sweep_id = sweep_features.get(rheo_sweep_num, {}).get('id', None)
+
+    if rheo_sweep_id is None:
+        raise Exception("Could not find id of rheobase sweep number %d." % rheo_sweep_num)
+
+    hero_sweep = cell_features["long_squares"]["hero_sweep"]
+    if hero_sweep is None:
+        raise Exception("Could not find hero sweep")
+
+    hero_sweep_num = hero_sweep["id"]
+    hero_sweep_id = sweep_features.get(hero_sweep_num, {}).get('id', None)
+
+    if hero_sweep_id is None:
+        raise Exception("Could not find id of hero sweep number %d." % hero_sweep_num)
+
+    # create a table of values
+    # this is a dictionary of ephys_features
+    base = cell_features["long_squares"]
+    ephys_features["rheobase_sweep_num"] = rheo_sweep_num
+    ephys_features["thumbnail_sweep_num"] = hero_sweep_num
+    ephys_features["vrest"] = nan_get(base, "v_baseline")
+    ephys_features["ri"] = nan_get(base, "input_resistance")
+
+    # change the base to hero sweep
+    base = cell_features["long_squares"]["hero_sweep"]
+    ephys_features["adaptation"] = nan_get(base, "adapt")
+    ephys_features["latency"] = nan_get(base, "latency")
+
+    # convert to ms
+    mean_isi = nan_get(base, "mean_isi")
+    ephys_features["avg_isi"] = (mean_isi * 1e3) if mean_isi is not None else None
+
+    # now grab the rheo spike
+    base = cell_features["long_squares"]["rheobase_sweep"]["spikes"][0]
+    ephys_features["upstroke_downstroke_ratio_long_square"] = nan_get(base, "upstroke_downstroke_ratio")
+    ephys_features["peak_v_long_square"] = nan_get(base, "peak_v")
+    ephys_features["peak_t_long_square"] = nan_get(base, "peak_t")
+    ephys_features["trough_v_long_square"] = nan_get(base, "trough_v")
+    ephys_features["trough_t_long_square"] = nan_get(base, "trough_t")
+    ephys_features["fast_trough_v_long_square"] = nan_get(base, "fast_trough_v")
+    ephys_features["fast_trough_t_long_square"] = nan_get(base, "fast_trough_t")
+    ephys_features["slow_trough_v_long_square"] = nan_get(base, "slow_trough_v")
+    ephys_features["slow_trough_t_long_square"] = nan_get(base, "slow_trough_t")
```

Comment:
Seems a bit repetitive maybe instead you could do something similar to:
```
features = ["upstroke_downstroke_ratio", "peak_v", ...]

postfix = '_long_square'
for ft in features:
    new_ft_name = '{}{}'.format(ft, postfix)
    ephys_features[new_ft_name] = nan_get(base, ft)
```

Reply:
@njmei Good suggestion. Done.